app/views/routes.py

```diff
@@ -0,0 +1,241 @@
+        }), 400
+
+    user = User(username, first_name, last_name, other_names, email, phone_number, password, is_admin)
+    user_list.append(user)
+
+    response = {
+      "status": 201,
+      "message": "User account created successfully",
+      "data": [user.to_dict()]
+    }
+    return jsonify(response), 201
+
+
+@redflag.route('/login', methods=['POST'])
+def login():
+  data = request.get_json(force=True)
+  username = data.get('username', None)
+  password = data.get('password', None)
+
+  for user in user_list:
+    if username == user.username and password == user.password:
+      return jsonify({
+        "status": 200,
+        "message": "Login successful",
+        "data": [user.to_dict()]
+      }), 200
+
+  return jsonify({
+    "status": 400,
+    "error": "Invaild username or password"
+  }), 400
+
+
+@redflag.route('/incident', methods=['POST'])
+def create_incident():
+  data = request.get_json(force=True)
+
+  created_by = data.get('created_by', None)
+  incident_type = data.get('incident_type', None)
+  location = data.get('location', None)
+  phone_number = data.get('phone_number', None)
+  status  = 'draft'
+  images = data.get('images', None)
+  videos = data.get('videos', None)
+  comment = data.get('comment', None)
+
+  incident = Incident(created_by, incident_type, location, status, images, videos, comment)
+  respo = validate_incident(incident)
+
+  if respo:
+    return jsonify(respo), 400
+
+  incident_list.append(incident)
+
+  response = {
+    "status": 201,
+    "message": "Incident record created successfully ",
+    "data": [incident.to_dict()]
+  }
+  
+  return jsonify(response), 201
+
+def validate_incident(incident):
+  errors = []
+  user_exists = False
+
+  if incident:
+    created_by = incident.created_by
+    incident_type = incident.incident_type
+    location = incident.location
+    status = incident.status
+    images = incident.images
+    videos = incident.videos
+
+    for user in user_list:
+      if created_by == user.username:
+        user_exists = True
+      break
+
+    if not user_exists:
+      errors.append({
+        "user_error": "User assigned doesnot exist"
+      })
+
+    if not location:
+      errors.append({
+        "location_error": "Location is missing"
+      })
+
+    if not created_by:
+      errors.append({
+        "error_created": "Please provide valid fields"
+      })
+
+    if not incident_type:
+      error.append({		
+        "incident_type": "Please specify the incident type"
+      })
+
+  else:
+    errors.append({
+      "error": "Please provide incident details"
+    })
+
+  if len(errors) > 0:
+    return {
+      "status": 400,
+      "errors": errors
+    }
+  else:
+   return None
+
+@redflag.route('/incident', methods=['GET'])
+def fetch_all_incident():
+  if incident_list:
+    return jsonify({
+        "status": 200,
+        "data": [incident.to_dict() for incident in incident_list]
+      })
+  return jsonify({
```

Comment:
simply return an empty list if the user has no records. Kindly refer to the challenge rubric for more info on this.

Reply:
`if not user_exists:
      errors.append({
        "user_error": "User assigned doesnot exist"
      })

     if not location:
      errors.append({
        "location_error": "Location is missing"
      })

     if not created_by:
      errors.append({
        "error_created": "Please provide valid fields"
      })

     if not incident_type:
      error.append({		
        "incident_type": "Please specify the incident type"
      })

   else:
    errors.append({
      "error": "Please provide incident details"
    })

   if len(errors) > 0:
    return {
      "status": 400,
      "errors": errors
    }`

Please keep the validation code in another file for easy readability and maintainability.